### ISSUE TITLE
SVG Stream from string

### ISSUE THREAD

**Issue by carloBarazzetta** [viniciusfbb/skia4delphi#2]
Hi, I'm trying to include skia4delphi as another SVG rendering engine into my project:
[https://github.com/EtheaDev/SVGIconImageList](https://github.com/EtheaDev/SVGIconImageList)
I've found how to load an SVG file into a TSKFileStream and then assigned it to FDOM: ISKSVGDOM.
But If I have the SVG content into a string, how can I prepare a valid stream to pass to FDOM?
I can use only TSKMemoryStream or you can add a TSKStringStream to simplify the work?
Thanks.
Carlo

**Comment by viniciusfbb** [comment]
Hi  @carloBarazzetta!

It is very simple, you can use the TSKStreamAdapter:
```delphi
var LStream: ISKStream := TSKStreamAdapter.Create(TStringStream.Create(FSource), True);
```

However there is a bug in the TSKStreamAdapter in the current version and it will not work in this version. The good news is that we've fixed this and other issues and will be released early next week in version 1.2.

**Comment by viniciusfbb** [comment]
Carlo, just a tip. You can implement the "fixed color" in svg with skia. See below:

```delphi
  procedure DrawWithDefaultColor(const ACanvas: ISKCanvas; const ADOM: ISKSVGDOM;
    const ADestRect: TRectF);
  var
    LSvgRect: TRectF;
  begin
    if ADestRect.IsEmpty or not Assigned(ADOM) then
      Exit;
    LSvgRect := TRectF.Create(TPointF.Create(0, 0), ADOM.Root.GetIntrinsicSize(TSizeF.Create(0, 0)));
    if LSvgRect.IsEmpty then
      Exit;
    ACanvas.Translate(ADestRect.Left, ADestRect.Top);
    ACanvas.Scale(ADestRect.Width / LSvgRect.Width, ADestRect.Height / LSvgRect.Height);
    FDOM.Render(ACanvas);
  end;

  procedure DrawWithFixedColor(const ACanvas: ISKCanvas; const ADOM: ISKSVGDOM;
    const ADestRect: TRectF; const AColor: TAlphaColor);
  var
    LSurface: ISKSurface;
    LImage: ISKImage;
    LPaint: ISKPaint;
    LSvgRect: TRectF;
  begin
    if ADestRect.IsEmpty or not Assigned(ADOM) then
      Exit;
    LSvgRect := TRectF.Create(TPointF.Create(0, 0), ADOM.Root.GetIntrinsicSize(TSizeF.Create(0, 0)));
    if LSvgRect.IsEmpty then
      Exit;
    LSurface := TSKSurface.MakeRaster(Round(ADestRect.Width), Round(ADestRect.Height));
    LSurface.Canvas.Clear(TAlphaColors.Null);
    LSurface.Canvas.Scale(ADestRect.Width / LSvgRect.Width, ADestRect.Height / LSvgRect.Height);
    FDOM.Render(LSurface.Canvas);
    LImage := LSurface.MakeImageSnapshot;
    LPaint := TSKPaint.Create;
    if FFillKind = TSvgBrushKind.Solid then
      LPaint.ColorFilter := TSKColorFilter.MakeBlend(FFillColor, TSkBlendMode.SrcIn);
    LPaint.Style := TSKPaintStyle.Fill;
    ACanvas.DrawImage(LImage, ADestRect.Left, ADestRect.Top, LPaint);
  end;
```

**Comment by carloBarazzetta** [comment]
Thank you, I'm trying to update the Basic demo with this method:
``` Delphi
procedure TfrmSVG.ShowText(const ASVGText: string);
var
  LSVGStream: ISKStream;
begin
  LSVGStream := TSKStreamAdapter.Create(
    TStringStream.Create(ASVGText, TEncoding.UTF8), True);
  FDOM := TSKSVGDOM.Make(LSVGStream);
  inherited Show;
  UpdateSVG;
end;
```

but when FDOM := TSKSVGDOM.Make(LSVGStream); the app hangs...
I'm doing something wrong?
Thanks.
Carlo

**Comment by viniciusfbb** [comment]
> but when FDOM := TSKSVGDOM.Make(LSVGStream); the app hangs...

Your code is correct, the problem is with TSKStreamAdapter bug, that I tell you:

> However there is a bug in the TSKStreamAdapter in the current version and it will not work in this version. The good news is that we've fixed this and other issues and will be released early next week in version 1.2.

**Comment by carloBarazzetta** [comment]
> However there is a bug in the TSKStreamAdapter in the current version and it will not work in this version. The good news is that we've fixed this and other issues and will be released early next week in version 1.2.

Sorry! I hadn't read!

**Comment by viniciusfbb** [comment]
Fixed in version 2.0.0
